command/server.go

```diff
@@ -79,6 +80,8 @@ type ServerCommand struct {
 	LogicalBackends    map[string]logical.Factory
 	PhysicalBackends   map[string]physical.Factory
 
+	ServiceDiscoverers map[string]sd.ServiceDiscoveryFactory
```

Comment:
I wonder if there is a better name we can use such as service registry and service registrar?